app/src/main/java/com/example/arc_exapmle/user/UserRepository.kt

```diff
@@ -17,21 +21,31 @@ class UserRepository(application: Application) {
         allUsers = userDao.getAllUsers()
     }
 
-    fun newUser(user: UserUI) {
+    suspend fun newUser(user: UserUI) {
 
-        NewUserAsyncTask(userDao).execute(user)
+        val userEntity = UserEntity(user.username, user.user_id)
+
+
+
+        userDao.insert(userEntity)
 
     }
 
     fun deleteUser(user: UserUI) {
 
-        DeleteUserAsyncTask(userDao).execute(user)
+        val userEntity = UserEntity(user.username, user.user_id)
+
+        CoroutineScope(Default).launch {
```

Comment:
in scope baz ham control nashode hast
be nazaram behtare az ye scope i ke control mishe estefade koni ke beshe cancel esh kard. ya age inja scope nadare aslan estefade nakon va bezaresh be ohdeye class haye balatar

Reply:
Inam hal shod